### PR TITLE
[front] chore: Handle permissions in resources

### PR DESCRIPTION
**Pull request by tdraier**
## Description

Check permissions directly in resources. Added results and error types

## Risk

Limited as we don't use groups everywhere yet

## Deploy Plan

deploy `front`

### CODE REVIEW

File: front/lib/resources/group_resource.ts
Comment (spolu) [inline front/lib/resources/group_resource.ts:39]: Feels a bit weird but could not come up with something better...

**Review by spolu — APPROVED** [review]
Looks great!

Based on this PR looks like a good direction :+1: 

Def need a second pair of eyes given criticality.

File: front/lib/resources/group_resource.ts
Comment (flvndvd) [inline front/lib/resources/group_resource.ts:41]: This should be proper instance `Error`.

File: front/lib/resources/types.ts
Comment (flvndvd) [inline front/lib/resources/types.ts:6]: This should be an instance Error.

File: front/lib/resources/group_resource.ts
Comment (flvndvd) [inline front/lib/resources/group_resource.ts:33]: `nit`
```suggestion
export type GroupErrorCode =
```

File: front/lib/resources/group_resource.ts
Comment (flvndvd) [inline front/lib/resources/group_resource.ts:218]: Why not calling `fetchByIds(auth, [id])`?

File: front/lib/resources/group_resource.ts
Comment (flvndvd) [inline front/lib/resources/group_resource.ts:420]: It's weird that this work, given that `GroupError` is defined as extending `Error` 🤔

File: front/lib/resources/group_resource.ts
Comment (flvndvd) [inline front/lib/resources/group_resource.ts:39]: I'm not sure we want to create one error type per resource. I'm worried that this scales pretty badly. We could probably create one base error class that accepts a code and a message, it could be enough, right?

File: front/lib/resources/types.ts
Comment (flvndvd) [inline front/lib/resources/types.ts:1]:
```suggestion
export type ResourceErrorCode =
```

File: front/lib/resources/types.ts
Comment (flvndvd) [inline front/lib/resources/types.ts:8]: `nit` but I'd code first then message.

File: front/pages/api/w/[wId]/groups/[gId]/index.ts
Comment (flvndvd) [inline front/pages/api/w/[wId]/groups/[gId]/index.ts:53]: This should become `groupRes` to follow the existing convention.

File: front/lib/resources/group_resource.ts
Comment (tdraier) [reply front/lib/resources/group_resource.ts:407]: Seems this code snapshot was prior to converting to Error type .. it has been replaced by `new Err(new GroupError`

**Comment by spolu** [comment]
Did you roll the sentry secrets?

**Review by spolu — APPROVED** [review]
Looks very reasonable

File: front/lib/resources/group_resource.ts
Comment (flvndvd) [reply front/lib/resources/group_resource.ts:387]: Just checked out your branch and it's not a proper error as well 🤔

**Review by flvndvd — APPROVED** [review]
LGTM once my last comment is fixed 🙃

File: front/lib/resources/group_resource.ts
Comment (tdraier) [reply front/lib/resources/group_resource.ts]: found it .. seems the type were compatible , bit was not a proper Error indeed

**Comment by tdraier** [comment]
> Did you roll the sentry secrets?

yes, they're revoked

**Comment by spolu** [comment]
Thanks thanks!

**Pull request merged**
